Tests/WitsmlExplorer.Api.Tests/Workers/CompareLogDataWorker.Tests.cs

```diff
@@ -0,0 +1,455 @@
+using System;
+using System.Collections.Generic;
+using System.Globalization;
+using System.Linq;
+using System.Threading.Tasks;
+
+using Microsoft.Extensions.Azure;
+using Microsoft.Extensions.Logging;
+
+using Moq;
+
+using Witsml;
+using Witsml.Data;
+using Witsml.Extensions;
+using Witsml.ServiceReference;
+
+using WitsmlExplorer.Api.Jobs;
+using WitsmlExplorer.Api.Jobs.Common;
+using WitsmlExplorer.Api.Models;
+using WitsmlExplorer.Api.Models.Reports;
+using WitsmlExplorer.Api.Services;
+using WitsmlExplorer.Api.Workers;
+
+using Xunit;
+
+namespace WitsmlExplorer.Api.Tests.Workers
+{
+    public class CompareLogDataWorkerTests
+    {
+        private readonly string _sourceWellUid = "sourcewelluid";
+        private readonly string _sourceWellboreUid = "sourcewellboreuid";
+        private readonly string _sourceLogUid = "sourceLogUid";
+        private readonly string _targetWellUid = "targetwelluid";
+        private readonly string _targetWellboreUid = "targetwellboreuid";
+        private readonly string _targetLogUid = "targetloguid";
+        private readonly DateTime _baseDateTime = DateTime.Parse("2023-09-28T08:10:00Z", CultureInfo.InvariantCulture);
+        private readonly Mock<IWitsmlClientProvider> _witsmlClientProvider;
+        private readonly Mock<IWitsmlClient> _witsmlSourceClient;
+        private readonly Mock<IWitsmlClient> _witsmlTargetClient;
+        private readonly Mock<ILogger<CompareLogDataJob>> _logger;
+        private readonly CompareLogDataWorker _worker;
+
+        public CompareLogDataWorkerTests()
+        {
+            _witsmlClientProvider = new Mock<IWitsmlClientProvider>();
+            _witsmlSourceClient = new Mock<IWitsmlClient>();
+            _witsmlTargetClient = new Mock<IWitsmlClient>();
+            _logger = new Mock<ILogger<CompareLogDataJob>>();
+            _witsmlClientProvider.Setup(provider => provider.GetSourceClient()).Returns(_witsmlSourceClient.Object);
+            _witsmlClientProvider.Setup(provider => provider.GetClient()).Returns(_witsmlTargetClient.Object);
+            _worker = new CompareLogDataWorker(_logger.Object, _witsmlClientProvider.Object);
+        }
+
+        [Fact]
+        public async Task CompareLogData_Equal_Logs()
```

Comment:
Maybe add a test similar to this one for equal time logs so that there is at least one test with and without errors for both log types?

Reply:
The log index type has been added as an argument to the test, fixed.